witness/src/Dscp/Witness/Logic/Processing.hs

```diff
@@ -64,23 +63,26 @@ applyBlockRaw toVerify block = do
             let actions = SD.constructCompositeActions @ps (SD.dmaAccessActions blockDBM)
                                                            (SD.dmaAccessActions stateDBM)
                 rwComp = do
-                  sblock <- SD.liftERoComp $ expandBlock block
-                  SD.applyBlockImpl toVerify blkStateConfig sblock
-                  sequence_ . fmap addTx . enumerate . bbTxs . bBody $ block
+                    sblock <- SD.liftERoComp $ expandBlock fees block
+                    SD.applyBlockImpl toVerify blkStateConfig sblock
+                    sequence_ . fmap addTx . enumerate . bbTxs . bBody $ block
+
                 addTx (idx, gTx) = SD.modifyRwCompChgAccum $ SD.ChangeSet $
                     M.singleton
                         (SD.inj . toGTxId . unGTxWitnessed $ gTx)
                         (SD.New . TxVal $ TxBlockRef (headerHash block) idx)
+
              in SD.runERwCompIO actions def rwComp >>=
                     \((), (SD.CompositeChgAccum blockCS_ stateCS_)) -> pure (blockCS_, stateCS_)
+
         proof <- runSdRIO $ SD.dmaApply stateDBM stateCS
         runSdRIO $ SD.dmaApply blockDBM blockCS
         pure proof
   where
     blockPrefixes = S.fromList [tipPrefix, blockPrefix, txPrefix]
 
 applyBlock :: WitnessWorkMode ctx m => Block -> m AvlProof
-applyBlock = applyBlockRaw True
+applyBlock = applyBlockRaw feeCoefficients True
 
 applyGenesisBlock :: WitnessWorkMode ctx m => m ()
-applyGenesisBlock = void $ applyBlockRaw False genesisBlock
+applyGenesisBlock = void $ applyBlockRaw noFees False genesisBlock
```

Comment:
Please elaborate on why did you eventually decide to pass fees directly instead of figuring out why block validator kicks in when we don't ask it to.

Reply:
> Genesis
> noFees

Reply:
When i pass `noFees` validator should not kick in. Why did you decide to add kostyl instead of figuring out the problem?